### ISSUE TITLE
Feature suggestion: Facility to define status workflows

### ISSUE THREAD

**Issue by danielduckworth** [tolgee/tolgee-platform#1429]
Hey, I've been checking out the project and it's great! The UI is easy to use and I'm going to take a look at the integration with Vue and i18n next.

The main thing I noticed was the lack of control over the translation workflow. The context I'm coming from (international Edtech) is slightly different to standard app interface localization. 

A lot of content organized as modules needs to be both localized or translated and go through multiple verification stages. I saw that [permission enhancements are on the roadmap](https://tolgee.io/roadmap). I think it make sense to consider status and workflow management while working on permissions since it works best when stages are defined by access rules.

Of course it doesn't make sense to have complex workflows for every element. However if translation keys could be organized into user defined collections then the status and workflow stages could be applied to collections of translation keys.

Perhaps this could be defined from the frontend or even programmatically from the structure of the translation keys.

Interested to see what you think.

**Comment by danielduckworth** [comment]
Done. Hopefully the timezone thing works out.

**Comment by jdimeo** [comment]
Is this still on the roadmap?

We also need some custom workflows, but nothing too fancy.
We want:
🟡 Translated -> 🟣 Reviewed, but has Typographic Error OR  🟣 Reviewed, but Incorrect Translation -> 🟢 Reviewed

This makes the most sense as a status because then it's _per translation_.  Tags only apply at the key level so it's much harder to filter keys down to a list like "Show me all reviewed translations that had a typographic or syntax error for `sw-KE` only"

These custom statuses can self identify which "real" status they should be handled as- my custom status should act as a reviewed status or a translated status or an outdated status... don't need to add custom logic to the backend (yet) to get significant value from just having a few different label options for filtering.

**Comment by jdimeo** [comment]
Might be related to #2209

**Comment by jdimeo** [comment]
Any progress on this? This is a high priority for us.

**Comment by JanCizmar** [comment]
Hey! Since the current states are used in too many places in the system and redesigning it would be extremely costly for us, I was thinking about adding some flags status flags, you would be able to configure in the settings separated from the clasic Translated/reviewed states. Would this suite your cases?

**Comment by jdimeo** [comment]
Yes, anything would work at this point... we trust your design! The big thing is being able to flag at the *translation* level (since tags are at the key level) different statuses that can be filtered.

**Comment by jdimeo** [comment]
Something like the red flag outdated or the machine translated logos, but we can manually create and assign them. Doesn't have to touch "real" statuses like reviewed and translated.

**Comment by jdimeo** [comment]
Still hoping for some movement on this. I know you have a lot of priorities but this is becoming a problem since it's been unaddressed for so long. thank you!

**Comment by JanCizmar** [comment]
Hey John! Thanks for letting us know. I will try to prioritize it! 🔝

We are currently implementing Glossaries and important AI enhancemets (so we have some place in the world where AI rules). I will try to propose it to the implementation cycle right after that, but I cannot promise it.

**Comment by JanCizmar** [comment]
Since we have released the translation labels, I am closing this issue for now. If this solution doesn't cover your use case, please create new one.

![Image](https://github.com/user-attachments/assets/1863b3a0-42c2-4275-a2aa-0288da669e0e)
